lucene/core/src/test/org/apache/lucene/search/MultiCollectorTest.java

```diff
@@ -163,4 +163,64 @@ public void testCacheScoresIfNecessary() throws IOException {
     reader.close();
     dir.close();
   }
+  
+  public void testScorerWrappingForTopScores() throws IOException {
+    Directory dir = newDirectory();
+    RandomIndexWriter iw = new RandomIndexWriter(random(), dir);
+    iw.addDocument(new Document());
+    iw.commit();
```

Comment:
no need to commit, the follow-up call to getReader() creates a NRT segment anyway